flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendMigrationTestBase.java

```diff
@@ -156,9 +158,10 @@ public void testKeyedValueStateRegistrationFailsIfNewStateSerializerIsIncompatib
 
             fail("should have failed");
         } catch (Exception expected) {
-            Assert.assertTrue(
-                    ExceptionUtils.findThrowable(expected, StateMigrationException.class)
-                            .isPresent());
+            assertThat(
+                            ExceptionUtils.findThrowable(expected, StateMigrationException.class)
+                                    .isPresent())
+                    .isTrue();
```

Comment:
I guess this could be replaced with usage of `assertThatThrownBy`
```java
 assertThatThrownBy(
                        () ->
                                testKeyedValueStateUpgrade(
                                        new ValueStateDescriptor<>(
                                                stateName, new TestType.V1TestTypeSerializer()),
                                        new ValueStateDescriptor<>(
                                                stateName,
                                                new TestType.IncompatibleTestTypeSerializer())))
                .isInstanceOf(StateMigrationException.class);
```